awx/sso/backends.py

```diff
@@ -327,31 +327,26 @@ def get_user(self, user_id):
         return super(SAMLAuth, self).get_user(user_id)
 
 
-def _update_m2m_from_groups(user, ldap_user, related, opts, remove=True):
+def _update_m2m_from_groups(ldap_user_group_dns, opts, remove=True):
     """
     Hepler function to update m2m relationship based on LDAP group membership.
     """
     should_add = False
     if opts is None:
-        return
+        return None
     elif not opts:
         pass
-    elif opts is True:
+    elif isinstance(opts, bool) and opts is True:
         should_add = True
     else:
         if isinstance(opts, str):
             opts = [opts]
-        for group_dn in opts:
-            if not isinstance(group_dn, str):
-                continue
-            if ldap_user._get_groups().is_member_of(group_dn):
-                should_add = True
+        # If any of the users groups matches any of the list options
+        if list(set.intersection(set(ldap_user_group_dns), set(opts))):
+            should_add = True
     if should_add:
-        user.save()
-        related.add(user)
-    elif remove and user in related.all():
-        user.save()
-        related.remove(user)
+        return True
+    return False
```

Comment:
I don't see any case where `should_add` would not be a boolean, so it would be simpler to just `return should_add`. Before, this wasn't totally viable because it took these side-effect actions.

Also, what does this method do anymore? The docstring was probably incomplete before, but seems borderline incorrect now.

Reply:
just saw your update, resolved :+1: